### PR TITLE
fix idr and criterion in solver benchmark

### PR DESCRIPTION
**Pull request by yhmtsai**
Small fix - add idr in solver_factory of benchmark.
is any parameter left for user input? I can also add it in this pr.

Add create_criterion to keep the consistent stopping criteria.

### CODE REVIEW

**Comment by yhmtsai** [comment]
format!

**Review by tcojean — APPROVED** [review]
LGTM.

File: benchmark/solver/solver.cpp [hunk 1]
```diff
@@ -43,12 +43,14 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <iostream>
 
 
+#include <ginkgo/core/preconditioner/isai.hpp>
+
+
```
Comment (thoasm) [inline benchmark/solver/solver.cpp:48]: Shouldn't this already be included in Line 33 with `<ginkgo/ginkgo.hpp>`, as it is a public Ginkgo header?

**Review by thoasm — APPROVED** [review]
LGTM!

File: benchmark/solver/solver.cpp [hunk 2]
```diff
@@ -152,6 +154,7 @@ const std::map<std::string, std::function<std::unique_ptr<gko::LinOpFactory>(
                    {"cg", create_solver<gko::solver::Cg<>>},
                    {"cgs", create_solver<gko::solver::Cgs<>>},
                    {"fcg", create_solver<gko::solver::Fcg<>>},
+                   {"idr", create_solver<gko::solver::Idr<>>},
```
Comment (upsj) [inline benchmark/solver/solver.cpp:157]: We could add the subspace dimension and kappa

File: benchmark/solver/solver.cpp (hunk 1 above)
Comment (upsj) [reply benchmark/solver/solver.cpp:48]: should in the worst case also be imported in preconditioners.hpp

**Review by upsj — APPROVED** [review]
LGTM!

**Comment by upsj** [comment]
check-format!

**Comment by yhmtsai** [comment]
rebase!

**Pull request merged**